auth0-actions/allow-github-organisations-and-map-saml.js

```diff
@@ -41,10 +41,42 @@ exports.onExecutePostLogin = async (event, api) => {
       const authorised = userOrganisations.data.map(organisation => organisation.login).some(organisation => allowedOrganisations.includes(organisation))
 
       if (authorised) {
+        const allowedDomain = JSON.parse(event.secrets.ALLOWED_DOMAINS)
+
+        // AWS requires the SAML nameID format to be an email address, which must
+        // exactly match an existing user in AWS SSO:
+        // https://docs.aws.amazon.com/singlesignon/latest/userguide/troubleshooting.html
+        api.samlResponse.setAttribute('http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress', `${event.user.nickname}${allowedDomain}`)
+        api.samlResponse.setAttribute('http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name', `${event.user.nickname}${allowedDomain}`)
+
+        // Set SAML attribute for the user's GitHub team memberships
+        const userTeamsResponse = await octokit.request('GET /user/teams').catch(error => api.access.deny(`Error retrieving teams from GitHub: ${error}`))
+        const userTeamSlugs = userTeamsResponse.data.map(team => team.slug)
+        api.samlResponse.setAttribute('https://aws.amazon.com/SAML/Attributes/PrincipalTag:github_team', userTeamSlugs.join(','))
+
         return // this empty return is required by auth0 to continue to the next action
       }
     }
   }
 
   return api.access.deny('Access denied.')
 }
+
+/*
+  This rule maps an authenticated user's information to the correct SAML attributes.
+*/
+
+exports.onExecutePostLogin = async (event, api) => {
```

Comment:
Can delete this now 🔥 Since all the logic is in the one action 💣 😁

Reply:
🤦‍♂️